client/src/components/formHelpers/form-validators.tsx

```diff
@@ -19,6 +19,27 @@ function isPathRoot(urlString: string): boolean {
 
 type Validator = (value: string) => React.ReactElement | null;
 
+// example link: https://learn.microsoft.com/en-us/training/achievements/learn.wwl.get-started-c-sharp-part-1.trophy?username=moT01&sharingId=E2EF453C1F9208B8
+export const isMicrosoftLearnLink = (value: string): boolean => {
+  let url;
+  try {
+    url = new URL(value);
+  } catch {
+    return false;
+  }
+
+  const correctDomain = url.hostname === 'learn.microsoft.com';
+  const correctPath = !!url.pathname.match(
+    /^\/[^/]+\/training\/achievements\//
+  );
```

Comment:
```suggestion
  const correctPath = !!url.pathname.match(
    /^\/[^\/]+\/training\/achievements\//
  );
```

There's an error in this regex. It does seem to work when you escape that. Maybe I didn't explain it enough, but the URL's are specific to the challenge. The URL for the first trophy challenge looks like this:

`https://learn.microsoft.com/en-us/training/achievements/learn.wwl.get-started-c-sharp-part-1.trophy?username=moT01&sharingId=E2EF453C1F9208B8`

The URL for the second trophy challenge looks like this:

`https://learn.microsoft.com/en-us/training/achievements/learn.wwl.get-started-c-sharp-part-2.trophy?username=moT01&sharingId=E2EF453C1F9208B8`

^ the only difference is the `part-#`. The same goes for the rest of the trophy challenges through `part-6` (all of them). It would be nice to validate that the `part-#` is correct for the specific challenge campers are trying to submit. I don't think it's entirely necessary for the MVP - here's a suggestion that's a little more specific though...

```suggestion
  const correctPath = !!url.pathname.match(
    /^\/[^\/]+\/training\/achievements\/learn\.wwl\.get-started-c-sharp-part-\d\.trophy$/
  );
```

Reply:
Do we plan to expand to more Microsoft courses in the future?

Reply:
I don't think it's necessary to escape the /. If it was then the tests would have failed.

That said, `learn\.wwl\.get-started-c-sharp-part-\d\.trophy$` makes sense. If we do expand the courses we can refine the regexes then.

Reply:
> I guess the regex does work. It is giving me a pattern error on regex101 - but if it works, it works.

I get the pattern error for the PHP compiler, but it goes away if I switch to the JavaScript compiler.

Reply:
That explains it.